core/shared/src/test/scala/fs2/StreamLawsSuite.scala

```diff
@@ -36,6 +36,20 @@ class StreamLawsSuite extends Fs2Suite with TestInstances {
         .eqv(x.attempt.compile.toVector, y.attempt.compile.toVector)
     )
 
+  test("tailRecM special test") {
+    val n = 50000
+    def advance(i: Int): Stream[IO, Either[Int, Int]] =
+      Stream.emit(if (i < n) Left(i + 1) else Right(i))
+
+    val res = Stream
+      .monadErrorInstance[IO]
+      .tailRecM[Int, Int](0)(advance)
+      .compile
+      .toList
+      .unsafeRunSync()(ioRuntime)
```

Comment:
This is where something strange is going on... When we use this `ioRuntime`, defined [here](https://github.com/typelevel/munit-cats-effect/blob/main/ce3/shared/src/main/scala/munit/CatsEffectSuite.scala#L30), this test work. However, if we use the `materializeRuntime` defined [here](https://github.com/typelevel/cats-effect/blob/series/3.x/testkit/shared/src/main/scala/cats/effect/testkit/TestInstances.scala#L260-L261), it hangs. 

However, it is not clear why stack-safety of `tailRecM` would be affected by this runtime.., 🤔

Reply:
I don't see any calls to `tickAll()` which seems related to tests hanging -- not sure why that hasn't been a problem until these changes though. It's also not related to stack safety...

Reply:
Ok, the tests were right. There was a stack overflow in there (lack of stack safety) in `tailRecM`, with the changes we had done. I have reverted changes to the tests.